src/mac/generic_handle_mac.c

```diff
@@ -330,14 +323,11 @@ libusbp_error * libusbp_generic_handle_set_timeout(
 
     libusbp_error * error = NULL;
 
-    if (error == NULL)
-    {
-        error = check_pipe_id(pipe_id);
-    }
+    error = check_pipe_id(pipe_id);
```

Comment:
I'm not in favor of this kind of change because it makes it harder to ever insert correct code that might cause errors above the call to check_pipe_id.  I understand you might value using `clang-tidy` more than I do and thus weigh the pros and cons differently.